src/core/parser.js

```diff
@@ -846,7 +846,11 @@ var Lexer = (function LexerClosure() {
             }
             strBuf.push(String.fromCharCode((x << 4) | x2));
           } else {
-            strBuf.push('#', String.fromCharCode(ch));
+            strBuf.push('#');
+            if (!Lexer.isSpace(ch)) {
+              strBuf.push(String.fromCharCode(ch));
+            }
+            this.stream.skip(-1); // Reset the stream position.
```

Comment:
I don't think we shall reset position here. Can you move `if (specialChars[ch]) { strBuf.push('#'); break; }` just after we read the character (line 841). 
By reading the [spec](http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#page=25), I can assume like that is not okay to write '#' without hex code (see "A NUMBER SIGN (23h) (#) in a name shall be written by using its 2-digit hexadecimal code (23), preceded by the NUMBER SIGN."-rule). Can we warn there are well? If we relaxing parsing here, can we also relax it for non-hex digit, e.g. just add '#' with bad character and warn?